tests/matrix_free/matrix_vector_stokes_1d.cc

```diff
@@ -0,0 +1,352 @@
+      {
+        velocity.reinit(cell);
+        velocity.read_dof_values(src, 0);
+        velocity.evaluate(EvaluationFlags::gradients | EvaluationFlags::values);
+        pressure.reinit(cell);
+        pressure.read_dof_values(src, dim);
+        pressure.evaluate(EvaluationFlags::values);
+
+        for (unsigned int q = 0; q < velocity.n_q_points; ++q)
+          {
+            Tensor<2, dim, vector_t> grad_u;
+            if constexpr (dim == 1)
+              grad_u[0] = velocity.get_gradient(q);
+            else
+              grad_u = velocity.get_gradient(q);
+
+            Tensor<1, dim, vector_t> val_u;
+            if constexpr (dim == 1)
+              val_u[0] = velocity.get_value(q);
+            else
+              val_u = velocity.get_value(q);
+
+            vector_t pres = pressure.get_value(q);
+            vector_t div  = -trace(grad_u);
+            pressure.submit_value(div, q);
+
+            // subtract p * I
+            for (unsigned int d = 0; d < dim; ++d)
+              grad_u[d][d] -= pres;
+
+            velocity.submit_gradient(grad_u, q);
+            velocity.submit_value(val_u, q);
+          }
+
+        velocity.integrate(EvaluationFlags::gradients |
+                           EvaluationFlags::values);
+        velocity.distribute_local_to_global(dst, 0);
+        pressure.integrate(EvaluationFlags::values);
+        pressure.distribute_local_to_global(dst, dim);
+      }
+  }
+
+
+  void
+  vmult(VectorType &dst, const VectorType &src) const
+  {
+    AssertDimension(dst.size(), dim + 1);
+    for (unsigned int d = 0; d < dim + 1; ++d)
+      dst[d] = 0;
+    data.cell_loop(&MatrixFreeTest<dim, degree_p, VectorType>::local_apply,
+                   this,
+                   dst,
+                   src);
+  };
+
+private:
+  const MatrixFree<dim, Number> &data;
+};
+
+
+
+template <int dim, int fe_degree>
+void
+test()
+{
+  Triangulation<dim> triangulation;
+  if constexpr (dim == 1)
+    GridGenerator::hyper_cube(triangulation);
+  else
+    create_mesh(triangulation);
+
+  if (fe_degree == 1)
+    triangulation.refine_global(4 - dim);
+  else
+    triangulation.refine_global(3 - dim);
+
+  FE_Q<dim>       fe_u(fe_degree + 1);
+  FE_Q<dim>       fe_p(fe_degree);
+  FESystem<dim>   fe(fe_u, dim, fe_p, 1);
+  DoFHandler<dim> dof_handler_u(triangulation);
+  DoFHandler<dim> dof_handler_p(triangulation);
+  DoFHandler<dim> dof_handler(triangulation);
+
+  MatrixFree<dim, double> mf_data;
+
+  AffineConstraints<double> constraints;
+
+  BlockSparsityPattern      sparsity_pattern;
+  BlockSparseMatrix<double> system_matrix;
+
+  BlockVector<double>         solution;
+  BlockVector<double>         system_rhs;
+  std::vector<Vector<double>> vec1, vec2;
+
+  dof_handler.distribute_dofs(fe);
+  dof_handler_u.distribute_dofs(fe_u);
+  dof_handler_p.distribute_dofs(fe_p);
+  DoFRenumbering::component_wise(dof_handler);
+
+  constraints.close();
```

Comment:
Is this really needed?
```suggestion
```

Reply:
I need it because https://github.com/dealii/dealii/blob/e04cad6ac71b6d8b8bd8b69b56f2b130428bc691/tests/matrix_free/matrix_vector_stokes_1d.cc#L281-L283 requires me to have a closed constraints object. I will remove the other code.